server/aesdsocket.c

```diff
@@ -326,29 +333,63 @@ void* threadfunc(void* thread_param)
         char *filename = "/var/tmp/aesdsocketdata";
         #endif
 
-        //Write obtained packet to the file
+        //Write obtained packet to the file IF it is not the "AESDCHAR_IOCSEEKTO:" string. If it is, use the case for the ioctl command below.
+        //If ioctl case, then do not write the string. 
+        //Extract the values of write_cmd and write_cmd_offset.
         int filetotest_fd = open(filename, O_RDWR | O_CREAT | O_APPEND, S_IWUSR | S_IRUSR | S_IWGRP | S_IRGRP | S_IROTH);
-        int ret_wr = write(filetotest_fd, packet_buffer, write_len);
-        file_size += ret_wr; //Adding to the current file
 
+                                /* IOCTL SUPPORT*/
+
+
+        int ioctl_str_len = 0; //Obtain length of the ioctl string
+
+        int is_ioctl = 0;   //Flag for checks while reading 
+        
+        ioctl_str_len = strlen(ioctl_format_str);   //Obtain the length of "AESD_IOCHARSEEKTO:"
+
+        if(strncmp(packet_buffer, ioctl_format_str, ioctl_str_len) == 0 )     //Check if the string received is the ioctl format str. strncmp returns 0 on success. 
+        {
+            struct aesd_seekto seekto;  //Init
+            seekto.write_cmd = 0;
+            seekto.write_cmd_offset = 0;
+            char *ioctl_buffer = packet_buffer; //Create a buffer locally to modify and extract write_cmd and write_cmd_offset
+
+            ioctl_buffer+=ioctl_str_len;    //Sets the ptr to the end of the string so we can extract only X&Y values.
+        
+            printf("The string is: %s and the length is %d", ioctl_buffer, ioctl_str_len);
+            //Credit:
+            //Extracting values from a formatted string - educative.io (sscanf) and ChatGPT
+            //Using sscanf to extract X&Y values as an alternative to atoi or similar methods. 
+            if(sscanf(ioctl_buffer, "%d,%d", &seekto.write_cmd, &seekto.write_cmd_offset) < 0) //Checking for EOF or other types of error
+            {
+                syslog(LOG_ERR, "Error with sscanf, errno: %d", errno);
```

Comment:
@suja8742 good idea to catch the sscanf failure here but what will happen in this case with respect to ioctl?  Is there any need/desire to call ioctl in this case?

Reply:
I should be breaking out of here to avoid ioctl(), not sure how I missed that. Thank you for pointing it out.